pkg/metrics/metrics.jsx

```diff
@@ -391,37 +400,62 @@ class CurrentMetrics extends React.Component {
                     const is_user = k.match(/^user.*user@\d+\.service.+/);
                     const label = k.replace(/.*\//, '').replace(/\.service$/, '');
                     // only keep cgroup basenames, and drop redundant .service suffix
-                    merged.push([label, v, is_user, false]);
+                    merged.push([label, v, is_user, false, k]);
                 }
                 // filter out podman containers
                 const matches = k.match(podmanCgroupRe);
                 if (matches && v) {
-                    // truncate to 12 chars like the podman output
-                    const containerid = matches.groups.containerid.substr(0, 12);
-                    const is_user = k.match(/^user.slice/);
-                    merged.push([containerid, v, is_user, true]);
+                    let is_user = false;
+                    let uid = '0';
+                    const containerid = matches.groups.containerid;
+                    const umatches = k.match(useridCgroupRe);
+                    if (umatches) {
+                        is_user = true;
+                        uid = umatches.groups.userid;
+                    }
+                    merged.push([containerid, v, is_user, true, uid]);
                 }
             });
             merged.sort((a, b) => b[1] - a[1]);
             return merged.slice(0, n);
         }
 
-        function cgroupClickHandler(name, is_user, is_container) {
+        function podNameInCache(podNameMapping, uid, containerid) {
+            if (uid in podNameMapping) {
+                return containerid in podNameMapping[uid];
+            } else {
+                return false;
+            }
+        }
+
+        function podName(podNameMapping, uid, containerid) {
+            if (podNameInCache(podNameMapping, uid, containerid)) {
+                return podNameMapping[uid][containerid];
+            } else {
+                return containerid.substr(0, 12);
+            }
+        }
+
+        function cgroupClickHandler(name, is_user, is_container, uid, podNameMapping) {
             if (is_container) {
-                cockpit.jump("/podman");
+                if (podNameInCache(podNameMapping, uid, name)) {
+                    cockpit.jump("/podman#/?name=" + podName(podNameMapping, uid, name));
+                } else {
+                    cockpit.jump("/podman");
+                }
             } else {
                 cockpit.jump("/system/services#/" + name + ".service" + (is_user ? "?owner=user" : ""));
             }
         }
 
-        function cgroupRow(name, value, is_user, is_container) {
+        function cgroupRow(name, value, is_user, is_container, uid, podNameMapping) {
```

Comment:
Again, if you make this a const arrow, you don't need to pass podNameMapping here at all, but just access it on this.state.

Reply:
This is now fixed by using `getCachedPodName` everywhere